src/engine/TextLimit.cpp

```diff
@@ -0,0 +1,162 @@
+//  Copyright 2024, University of Freiburg,
+//                  Chair of Algorithms and Data Structures.
+//  Author: Nick Göckel <nick.goeckel@students.uni-freiburg.de>
+
+#include "./TextLimit.h"
+
+// _____________________________________________________________________________
+TextLimit::TextLimit(QueryExecutionContext* qec, const size_t n,
+                     std::shared_ptr<QueryExecutionTree> child,
+                     const ColumnIndex& textRecordColumn,
+                     const vector<ColumnIndex>& entityColumns,
+                     const vector<ColumnIndex>& scoreColumns)
+    : Operation(qec),
+      qec_(qec),
+      n_(n),
+      child_(std::move(child)),
+      textRecordColumn_(textRecordColumn),
+      entityColumns_(entityColumns),
+      scoreColumns_(scoreColumns) {}
+
+// _____________________________________________________________________________
+ResultTable TextLimit::computeResult() {
+  shared_ptr<const ResultTable> childRes = child_->getResult();
+
+  if (n_ == 0) {
+    return {IdTable(childRes->width(), getExecutionContext()->getAllocator()),
+            resultSortedOn(), childRes->getSharedLocalVocab()};
+  }
+
+  IdTable idTable =
+      childRes->idTable().clone();  // QUESTION: is there a cheaper solution?
+  // Sort the table by the entity column, then the score column, then the text
+  // column.
+
+  auto compareScores = [this](const auto& lhs, const auto& rhs) {
+    size_t lhsScore = 0;
+    size_t rhsScore = 0;
+    for (size_t i = 0; i < scoreColumns_.size(); ++i) {
+      lhsScore += lhs[scoreColumns_[i]].getInt();
+      rhsScore += rhs[scoreColumns_[i]].getInt();
+    }
+    if (lhsScore > rhsScore) {
+      return 1;
+    } else if (lhsScore < rhsScore) {
+      return -1;
+    } else {
+      return 0;
+    }
+  };
+
+  auto compareEntities = [this](const auto& lhs, const auto& rhs) {
+    for (size_t i = 0; i < entityColumns_.size(); ++i) {
+      if (lhs[entityColumns_[i]] > rhs[entityColumns_[i]]) {
+        return 1;
+      } else if (lhs[entityColumns_[i]] < rhs[entityColumns_[i]]) {
+        return -1;
+      }
+    }
+    return 0;
+  };
+
+  // Runtime: O(n log n)
+  std::ranges::sort(idTable, [this, compareScores, compareEntities](
+                                 const auto& lhs, const auto& rhs) {
+    return compareEntities(lhs, rhs) == 1 ||
+           (compareEntities(lhs, rhs) == 0 &&
+            (compareScores(lhs, rhs) == 1 ||
+             (compareScores(lhs, rhs) == 0 &&
+              (lhs[textRecordColumn_] > rhs[textRecordColumn_]))));
+  });
```

Comment:
The sorting could still be done in a different way, 
I will think of something.
For now add a 
`// TODO<joka921> Let the `SORT` class handle this. This requires descending sorting for positive integers though.`

Or maybe you can just use the `ORDER BY` class here, and actually in the constructor set the child to be an `ORDER BY` of the input, then you have the descending sorting without this whole code bloat.